notification/src/handler/notification/notification.handler.js

```diff
@@ -336,18 +391,41 @@ async function calculateUpdateActionsForPayment(payment, notification, logger) {
         ),
       )
     }
-    const paymentKey = payment.key
-    const newPspReference =
-      notificationRequestItem.originalReference || pspReference
-    if (newPspReference && newPspReference !== paymentKey) {
-      updateActions.push({
-        action: 'setKey',
-        key: newPspReference,
-      })
+
+    if (notificationRequestItem.success) {
+      const paymentKey = payment.key
+      const newPspReference =
+        notificationRequestItem.originalReference || pspReference
+      if (newPspReference && newPspReference !== paymentKey) {
+        updateActions.push({
+          action: 'setKey',
+          key: newPspReference,
+        })
+      }
     }
 
-    if (transactionType === 'Authorization' && transactionState === 'Success'
-        && !notificationRequestItem.operations.includes('CAPTURE')) {
+    if (
+      transactionType === 'Authorization' &&
+      transactionState === 'Success' &&
+      !notificationRequestItem.operations.includes('CAPTURE')
+    ) {
+      updateActions.push(
+        getAddTransactionUpdateAction({
+          timestamp: convertDateToUTCFormat(eventDate, logger),
+          type: 'Charge',
+          state: 'Success',
+          amount: notificationRequestItem.amount.value,
+          currency: notificationRequestItem.amount.currency,
+          interactionId: pspReference,
+        }),
+      )
+    }
```

Comment:
isn't this block (lines 407-422) exactly the same as block in lines 424-440?
maybe was unnecessarily added during merge conflicts solving?

Reply:
Yes I probably duplicated it during merge, thanks